src/main/scala/core/database/DatabaseActor.scala

```diff
@@ -1,36 +1,24 @@
-package database
+package core.database
 
 import java.sql.{Connection, DriverManager, ResultSet}
 
-import akka.actor.{Actor, ActorLogging, ActorRef, PoisonPill, Props}
+import akka.actor.{Actor, ActorLogging, ActorRef, PoisonPill}
 import akka.stream._
-import database.DatabaseActor.QueryResult
-import main.PropsWithName
 import org.h2.jdbc.JdbcSQLException
-import utils.Configuration
+import core.utils.Configuration
+import core.utils.Message
+import core._
 
 import scala.collection.mutable.ArrayBuffer
 import scala.util.{Failure, Success, Try}
 
 /**
   * Created by igor on 25/05/17.
   */
-object DatabaseActor {
-  case object Shutdown
-
-  case class QueryDB(reqId: Int, query: String, update: Boolean = false)
```

Comment:
The Akka team also recommends to declare the messages an Actor can receive in the [companion object](https://doc.akka.io/docs/akka/current/actors.html?language=scala#recommended-practices)

Reply:
You mean it is good because of Exhaustiveness Checking?